client/rest-client/src/main/java/org/infinispan/client/rest/RestEntity.java

```diff
@@ -137,4 +137,12 @@ static RestEntity create(File file) {
       }
       return RestEntity.create(MediaType.APPLICATION_OCTET_STREAM, file);
    }
+
+   static RestEntity create(String s) {
+      return switch (s.trim().charAt(0)) {
+         case '{' -> RestEntity.create(MediaType.APPLICATION_JSON, s);
+         case '<' -> RestEntity.create(MediaType.APPLICATION_XML, s);
+         default -> RestEntity.create(MediaType.TEXT_PLAIN, s);
```

Comment:
Hmm my comment from yesterday seems to have been lost :thinking: 

What about if we are passing yaml?

The following fails: `create cache distcache "distributedCache: ~"`

I know it's hard to guess yaml, so maybe we should introduce an optional `media-type` flag?

So it would be  `create cache distcache "distributedCache: ~" --media-type=<JSON|XML|YAML>`

Reply:
There was an error in `ConfigurationReader`

Reply:
I'm still getting an error, but it's not very informative:

```
[df0dba52a5e5-62733@cluster//containers/default]> create cache yaml "distributedCache: ~"
Error: 
```

Reply:
I've added tests to `CliIT`. They work locally

Reply:
Argh, I wasn't engaging my brain. I was running an old server in docker to test interactively ... obviously that doesn't have the fixes you've added to the backend :man_facepalming: